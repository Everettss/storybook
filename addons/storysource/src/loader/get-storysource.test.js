import fs from 'fs';
import path from 'path';
import getStorySource from './get-storysource';

describe('get-storysource', () => {
  describe('positive', () => {
    const mockFilePath = './__mocks__/inject-decorator.stories.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');
    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'javascript',
    });

    it('calculates "adds" map', () => {
      expect(result.addsMap).toMatchSnapshot();
    });
  });

  describe('positive - angular', () => {
    const mockFilePath = './__mocks__/inject-decorator.angular-stories.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');
    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'typescript',
    });

    it('calculates "adds" map', () => {
      expect(result.addsMap).toMatchSnapshot();
    });
  });

  describe('positive - ts', () => {
    const mockFilePath = './__mocks__/inject-decorator.ts.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');
    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'typescript',
    });

    it('calculates "adds" map', () => {
      expect(result.addsMap).toMatchSnapshot();
    });
  });

  describe('stories with ugly comments', () => {
    const mockFilePath = './__mocks__/inject-decorator.ugly-comments-stories.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');
    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'javascript',
    });

    it('should delete ugly comments from the generated story source', () => {
      expect(result.storySource).toMatchSnapshot();
    });
  });

  describe('stories with ugly comments in ts', () => {
    const mockFilePath = './__mocks__/inject-decorator.ts.ugly-comments-stories.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');
    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'typescript',
    });

    it('should delete ugly comments from the generated story source', () => {
      expect(result.storySource).toMatchSnapshot();
    });
  });

  it('will not return addsMap when there are no "storiesOf" functions', () => {
    const mockFilePath = './__mocks__/inject-decorator.no-stories.txt';
    const source = fs.readFileSync(mockFilePath, 'utf-8');

    const result = getStorySource(source, path.resolve(__dirname, mockFilePath), {
      parser: 'javascript',
    });

    expect(result.addsMap).toEqual({});
  });
});
