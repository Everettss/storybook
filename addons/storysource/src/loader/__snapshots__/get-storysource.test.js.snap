// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-storysource positive - angular calculates "adds" map 1`] = `
Object {
  "Custom|ng-content@Default": Object {
    "endLoc": Object {
      "col": 2,
      "line": 15,
    },
    "startLoc": Object {
      "col": 43,
      "line": 10,
    },
  },
}
`;

exports[`get-storysource positive - ts calculates "adds" map 1`] = `
Object {
  "ngrx|Store@With component": Object {
    "endLoc": Object {
      "col": 3,
      "line": 32,
    },
    "startLoc": Object {
      "col": 7,
      "line": 28,
    },
  },
}
`;

exports[`get-storysource positive calculates "adds" map 1`] = `
Object {
  "Addons|Info.Decorator@Use Info as story decorator": Object {
    "endLoc": Object {
      "col": 73,
      "line": 137,
    },
    "startLoc": Object {
      "col": 7,
      "line": 137,
    },
  },
  "Addons|Info.GitHub issues@#1814": Object {
    "endLoc": Object {
      "col": 4,
      "line": 152,
    },
    "startLoc": Object {
      "col": 2,
      "line": 146,
    },
  },
  "Addons|Info.Markdown@Displays Markdown in description": Object {
    "endLoc": Object {
      "col": 96,
      "line": 44,
    },
    "startLoc": Object {
      "col": 2,
      "line": 43,
    },
  },
  "Addons|Info.Options.TableComponent@Use a custom component for the table": Object {
    "endLoc": Object {
      "col": 41,
      "line": 130,
    },
    "startLoc": Object {
      "col": 2,
      "line": 127,
    },
  },
  "Addons|Info.Options.header@Shows or hides Info Addon header": Object {
    "endLoc": Object {
      "col": 41,
      "line": 60,
    },
    "startLoc": Object {
      "col": 2,
      "line": 56,
    },
  },
  "Addons|Info.Options.inline@Inlines component inside story": Object {
    "endLoc": Object {
      "col": 41,
      "line": 52,
    },
    "startLoc": Object {
      "col": 2,
      "line": 48,
    },
  },
  "Addons|Info.Options.propTables@Shows additional component prop tables": Object {
    "endLoc": Object {
      "col": 41,
      "line": 76,
    },
    "startLoc": Object {
      "col": 2,
      "line": 72,
    },
  },
  "Addons|Info.Options.propTablesExclude@Exclude component from prop tables": Object {
    "endLoc": Object {
      "col": 4,
      "line": 89,
    },
    "startLoc": Object {
      "col": 2,
      "line": 80,
    },
  },
  "Addons|Info.Options.source@Shows or hides Info Addon source": Object {
    "endLoc": Object {
      "col": 41,
      "line": 68,
    },
    "startLoc": Object {
      "col": 2,
      "line": 64,
    },
  },
  "Addons|Info.Options.styles@Extend info styles with an object": Object {
    "endLoc": Object {
      "col": 43,
      "line": 108,
    },
    "startLoc": Object {
      "col": 4,
      "line": 94,
    },
  },
  "Addons|Info.Options.styles@Full control over styles using a function": Object {
    "endLoc": Object {
      "col": 43,
      "line": 123,
    },
    "startLoc": Object {
      "col": 4,
      "line": 111,
    },
  },
  "Addons|Info.React Docgen@Comments from Flow declarations": Object {
    "endLoc": Object {
      "col": 85,
      "line": 22,
    },
    "startLoc": Object {
      "col": 4,
      "line": 19,
    },
  },
  "Addons|Info.React Docgen@Comments from PropType declarations": Object {
    "endLoc": Object {
      "col": 79,
      "line": 16,
    },
    "startLoc": Object {
      "col": 4,
      "line": 13,
    },
  },
  "Addons|Info.React Docgen@Comments from component declaration": Object {
    "endLoc": Object {
      "col": 70,
      "line": 28,
    },
    "startLoc": Object {
      "col": 4,
      "line": 25,
    },
  },
}
`;

exports[`get-storysource stories with ugly comments in ts should delete ugly comments from the generated story source 1`] = `
"import React from 'react';

@Component({
  selector: 'storybook-comp-with-store',
  template: '<div>{{this.getSotreState()}}</div>',
})
class WithStoreComponent {
  private store: Store<any>;

  constructor(store: Store<any>) {
    this.store = store;
  }

  getSotreState(): string {
    return this.store === undefined ? 'Store is NOT injected' : 'Store is injected';
  }
}

import { storiesOf } from '@storybook/react';

const x = 0;

storiesOf('Foo', module).add('bar', () => <div>baz</div>);

/*
 This is actually a good comment that will help
 users to understand what's going on here.
*/
"
`;

exports[`get-storysource stories with ugly comments should delete ugly comments from the generated story source 1`] = `
"import React from 'react';

import { storiesOf } from '@storybook/react';

const x = 0;

storiesOf('Foo', module).add('bar', () => <div>baz</div>);

/*
 This is actually a good comment that will help
 users to understand what's going on here.
*/
"
`;
